pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -709,6 +709,13 @@ func buildSidecarOutboundTCPListenerOptsForPortOrUDS(listenerMapKey *string,
 	if len(listenerOpts.bind) == 0 {
 		svcListenAddress := listenerOpts.service.GetAddressForProxy(listenerOpts.proxy)
 		svcExtraListenAddress := listenerOpts.service.GetExtraAddressesForProxy(listenerOpts.proxy)
+		// Override the svcListenAddress, using the proxy ipFamily, for cases where the ipFamily cannot be detected easily.
+		// For example: due to the possibility of using hostnames instead of ips in ServiceEntry,
+		// it is hard to detect ipFamily for such services.
+		if listenerOpts.service.Attributes.ServiceRegistry == provider.External && listenerOpts.proxy.IsIPv6() &&
+			svcListenAddress == constants.UnspecifiedIP {
```

Comment:
Hi @kfaseela, `svcListenAddress` is set to `constants.UnspecifiedIP` in [here](https://github.com/istio/istio/blob/08b907f7e46784e042fd05f4d397b42b17464e8e/pilot/pkg/serviceregistry/serviceentry/conversion.go#LL200C29-L200C29), and it's used in many places for service entry. Therefore, IMO, `constants.UnspecifiedIP` is symbolic rather than practical for service entry.  Is it possible for us to change it and use a special constant only for service entry? such as `None` or `Wildcard`?